### ISSUE TITLE
🛠️ Run an automated test in Azure

### ISSUE THREAD

**Issue by petertgiles** [GCTC-NTGC/gc-digital-talent#7999]
## 🛠️ Tooling

We recently had a few near misses with issues found in UAT in Azure before deployment.  There was the auth issue where the first request after logging in failed.  And then the database upgrade with the misconfigured extension.  I was thinking that it would be handy if we could run a batch of tests like ~Cypress~ Playwright against the dev or uat instance in Azure.

If the dev vertical was open to the Internet I wonder if it would be possible to open Cypress on a dev's machine and point it at the Azure instance?

## ✅ Acceptance Criteria

- [ ] How do we manually  run a batch of automated tests against an instance running in Azure Dev 

## Timebox 
3 days

**Comment by brindasasi** [comment]
IMTD mentioned that since the tests are invoked by Azure managed pipelines , it uses various ips and F5 is blocking them.
The suggested solution is to use self hosted VMs.
I've asked them to install the necessary apps into T1_Dev_Management VM. 
Waiting on that .............

**Comment by brindasasi** [comment]
Yayy.. some one is actually working on this although he is waiting on the security approval on his below assessment 

> DASTPIER[St-Pierre, Daniel](https://jira.tbs-sct.gc.ca/secure/ViewProfile.jspa?name=DASTPIER) logged work - 2024-03-19 1:20 PM
Time Spent:	1 hour
 	
Assessment of what would be installed on self-hosted agent

reference pipeline: https://github.com/GCTC-NTGC/gc-digital-talent/blob/7999-integrate-playwright-azure/infrastructure/azure-pipelines-dev.yml

 

PHP and Composer would be install on self-hosted agent vm.
Node/npm would be installed on self-hosted agent vm

pipeline execution of script "deploy.sh" specified node v20.11.0
 

Workload:

php 8.2
info: https://wiki.crowncloud.net/?How_to_install_php_8_2_on_Ubuntu_20_04
agent VM mostly likely needs access to:
keyserver.ubuntu.com (tcp: 11371)
ppa.launchpadcontent.net/ondrej/php/ubuntu/ (tcp: 443)
php composer
info: https://getcomposer.org/
agent VM most likely needs access to:
repo.packagist.org (tcp: 443)
api.github.com (tcp: 443)
github.com (tcp: 443)
codeload.github.com (tcp: 443)
node/npm (v20.11.0 as defined in project script: https://github.com/GCTC-NTGC/gc-digital-talent/blob/221fc585929da8a5cec095ddb10cbe3114229ac9/infrastructure/bin/deploy.sh#L52)
download: https://nodejs.org/dist/v20.11.0/node-v20.11.0-linux-x64.tar.xz
manual install bare info:
https://www.tencentcloud.com/document/product/213/34812
https://stackoverflow.com/questions/65775227/how-can-i-install-nodejs-manually-in-linux-from-terminal
 

 

Assessment of Devops Artifacts Feed

 

Current npm feed already exists at organization scope: TBS_Custom_Npm_Feed
May need to adjust feed permissions to allow TC developers access to read/write

Adjust affected devops users' VMs and project code according to feed connection steps: https://dev.azure.com/tbs-sct/Talent%20Cloud/_packaging?_a=connect&feed=TBS_Custom_Npm_Feed

**Comment by brindasasi** [comment]
**Azure Dev Pipeline :** 
GCTC-NTGC.gc-digital-talent.dev - 

This pipeline already is set to run with dev dependencies like faker. 
Now we have split it into 2 stages 

**Build Stage** - 
Code changes are built and an artifact has been created. This build agent has been hosted by Azure. 

**Test Stage** 
In this stage , this artifact has been feed into an self hosted build agent and tests can run against either Dev or UAT sites those are under VPN. 

This infrastructure has been set up and ready to use. 

Now the playwright tests are running but needs some refactoring to use the actual GCKey Sign in instead of the mock auth. 

We create some test accounts as we did on UAT for UX testing and feed into these tests. 

Only thing we are waiting on this from IMTD is to provide the easy way of getting the test logs. Currently it is under self hosted vm.

Daniel is working on to get this on to a storage account so we can download it easily.

**Comment by brindasasi** [comment]
So all the infrastructure is now set up and I try my best to list the details in the PR.
I've created below ticket for adding GCKey logins to the tests
https://github.com/GCTC-NTGC/gc-digital-talent/issues/10217

**Comment by brindasasi** [comment]
This is not a spike anymore.
This is actually delivers a pipeline in Azure that can tun the playwright tests
